xbmc/cores/VideoPlayer/DVDCodecs/Video/MMALCodec.h

```diff
@@ -80,7 +80,7 @@ class CMMALVideo : public CDVDVideoCodec
   virtual void Reset(void);
   virtual bool GetPicture(DVDVideoPicture *pDvdVideoPicture);
   virtual bool ClearPicture(DVDVideoPicture* pDvdVideoPicture);
-  virtual unsigned GetAllowedReferences() { return 3; }
+  virtual unsigned GetAllowedReferences() { return NUM_BUFFERS; }
```

Comment:
you also return NUM_BUFFERS here: https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/MMALRenderer.cpp#L59

Is that your intent? This results in using 6 buffers.